go/worker/common/committee/keymanager.go

```diff
@@ -317,3 +410,152 @@ func newKeyManagerNodeTracker(p2p p2p.Service, consensus consensus.Backend, keym
 		logger:       logging.GetLogger("worker/common/committee/keymanager/nodetracker"),
 	}
 }
+
+// peerFeedbackNode represents a single node in the peer feedback cache.
+type peerFeedbackNode struct {
+	// requestID is the ID of the request.
+	requestID uint64
+	// feedback holds the peer feedback stored in this node.
+	feedback rpc.PeerFeedback
+	// timestamp is the time when the feedback was added to the cache.
+	timestamp time.Time
+	// prev is a pointer to the previous node in the cache.
+	prev *peerFeedbackNode
+	// next is a pointer to the next node in the cache.
+	next *peerFeedbackNode
+}
+
+// peerFeedbackCache is a cache that holds the most recent unused peer feedback.
+type peerFeedbackCache struct {
```

Comment:
Consider `container/list` and/or `go/common/cache/lru`.

Reply:
Used `lru`, stupid me :facepalm: